tabs/ships.py

```diff
@@ -0,0 +1,273 @@
+    ship_scroll = ttk.Scrollbar(
+        ship_summary, orient=tk.VERTICAL, command=ship_view.yview
+    )
+    ship_scroll.grid(column=1, row=0, sticky=tk.NS)
+    ship_view["yscrollcommand"] = ship_scroll.set
+
+    ship_summary.columnconfigure(0, weight=1)
+    ship_summary.rowconfigure(0, weight=1)
+
+    widgets['CurrentShips'] = ship_view
+
+    #Drop down menu
+
+    variable = tk.StringVar(parent)
+    variable.set("Double click row to select ship in table") # default value
+
+    dropDown = ttk.OptionMenu(ships_page, variable)
+    dropDown.option_clear
+    dropDown.grid(row=1, sticky=(tk.NSEW),padx=20,pady=20)
+
+    widgets['Dropdown'] = dropDown
+
+    parent.columnconfigure(0, weight=1)
+    parent.rowconfigure(0, weight=1)
+    parent.columnconfigure(1, weight=1)
+    parent.rowconfigure(1, weight=1)
+
+    ships_page.grid(row=0, column=0, sticky=tk.NSEW)
+    ship_summary.grid(row=0, column=1, sticky=tk.NSEW)
+
+    #credits refresh
+    global creditsCurrent
+    creditsCurrent = tk.StringVar(parent)
+    try:
+        response = requests.get(
+            ACCOUNT, params={"token": trader_token.get()}, **proxy_workaround
+        )
+        if response.status_code == 200:
+            result = response.json()
+            creditsCurrent.set(result['user']['credits'])
+    except ConnectionError as ce:
+        print("Failed:", ce)
+
+    label = ttk.Label(ships_page, text=creditsCurrent.get())
+    widgets['CreditsLabel'] = label
+    label.grid(row=2, column=0, sticky=tk.NSEW)
+
+
+
+def purchaseMenu(event):
+    data = event.split(":")
+
+    try:
+        response = requests.get(
+            ACCOUNT, params={"token": trader_token.get()}, **proxy_workaround
+        )
+        if response.status_code == 200:
+            result = response.json()
+            credits = result['user']['credits']
+            if credits < int(data[1][1:]):
+                tkinter.messagebox.showerror(title='Insufficient Credits',message='You do not have enough credits to purchase this ship')
+            else:
+                try:
+                    response = requests.get(
+                    MY_SHIPS, params={"token": trader_token.get()}, **proxy_workaround
+                    )
+                    if response.status_code == 200:
+                        result = response.json()
+                        locationAccepted = 0
+                        for ship in result['ships']:
+                            if ship['location'] == data[0]:
+                                locationAccepted = 1
+                                selected = widgets['ShipsTable'].selection()[0]
+                                shipType = widgets['ShipsTable'].item(selected,"values")[5]
+                                print(trader_token.get(),ship['location'],shipType)
+                                if tkinter.messagebox.askyesno(title='Confirm Purchase',message='Do you want to purchase %s for %s credits'%(shipType,int(data[1][1:]))):
```

Comment:
f-strings are preferred over old-style formatting (and the performance is better, I believe)